### DISCUSSION TITLE
301 Moved Permanently

### DISCUSSION THREAD

**Discussion by salarizadi** [Moddable-OpenSource/moddable#1168] [Q&A]
Hello.

I want to make an http get request to a site, but unfortunately I cannot receive information from that site.

and shows the 301 error :
```html
<!DOCTYPE html>
<html style="height:100%">
<head>
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
<title> 301 Moved Permanently

</title></head>
<body style="color: #444; margin:0;font: normal 14px/20px Arial, Helvetica, sans-serif; height:100%; background-color: #fff;">
<div style="height:auto; min-height:100%; ">     <div style="text-align: center; width:800px; margin-left: -400px; position:absolute; top: 30%; left:50%;">
        <h1 style="margin:0; font-size:150px; line-height:150px; font-weight:bold;">301</h1>
<h2 style="margin-top:20px;font-size: 30px;">Moved Permanently

</h2>
<p>The document has been permanently moved.</p>
</div></div></body></html>
```

My code : 
```js
import Net from "net";
import WiFi from "wifi/connection";
import {Server, Request} from "http"
import SecureSocket from "securesocket";

let request = new Request({
	host: "", path: "/", response: String
});
request.callback = function (message, value, etc) {
	if (Request.header === message)
		trace(`${value}: ${etc}\n`);
	else if (Request.responseComplete === message)
		trace(value + "\n");
}

```

**Comment by phoddie** [comment]
It looks like you are making an `http` request and the site is responding with a redirect (301). It is probably redirecting you to an `https` URL. It looks like your code is tracing the HTTP response headers from the server. Do you see a `location` header there with the destination address?

Note that the HTTP Request object does **not** automatically resolve HTTP redirects for you.

**Reply by salarizadi** [reply]
yes it seems to go from http to https but after doing that it doesn't show any forwarding address anywhere and even i use websocket https it doesn't go anywhere and it doesn't show any information from it. Even from, for example, gtihub.com, I want to get the information in the form of String, this is not possible and the Request does not work at all. I don't know how to do this. If there is a complete code sample to solve this problem, please send it. I would appreciate it.

**Reply by phoddie** [reply]
I can probably be of more help if I can understand what you are trying to do. Would you explain? If it is just to make an HTTPS GET then start with the [httpsget](https://github.com/Moddable-OpenSource/moddable/blob/public/examples/network/http/httpsget/main.js) example instead of httpget.
